nixos/modules/programs/nano.nix

```diff
@@ -32,20 +33,15 @@ in
     };
   };
 
-  ###### implementation
+  config = lib.mkIf cfg.enable {
+    environment = {
+      etc.nanorc.text = lib.concatStringsSep "\n" cfg.nanorc;
+      systemPackages = [ cfg.package ];
+    };
 
-  config = lib.mkIf (cfg.nanorc != "" || cfg.syntaxHighlight) {
-    environment.etc.nanorc.text = lib.concatStringsSep LF (
-      ( lib.optionals cfg.syntaxHighlight [
-          "# The line below is added because value of programs.nano.syntaxHighlight is set to true"
-          ''include "${pkgs.nano}/share/nano/*.nanorc"''
-          ""
-      ])
-      ++ ( lib.optionals (cfg.nanorc != "") [
-        "# The lines below have been set from value of programs.nano.nanorc"
-        cfg.nanorc
-      ])
-    );
+    programs.nano.nanorc = lib.mkIf cfg.syntaxHighlight ''
```

Comment:
https://github.com/NixOS/nixpkgs/blob/883fd35c940aa304654f9bc796e6d53ad627a9c8/nixos/modules/config/system-path.nix#L45
nano is a defaultPackage, so this could break some people's setups

Reply:
I got an idea, system-path.nix we should check if nano is in defaultPackages and then enable based on that

Reply:
```patch
commit 7f4373ca25970be0438a752389073ef34c23a06a
Author: Artturin <Artturin@artturin.com>
Date:   Sun Jul 23 00:01:08 2023 +0300

    nixos/system-path: enable nano module based on defaultPackages

diff --git a/nixos/modules/config/system-path.nix b/nixos/modules/config/system-path.nix
index e8bbeac4f720..3a39f06002f1 100644
--- a/nixos/modules/config/system-path.nix
+++ b/nixos/modules/config/system-path.nix
@@ -143,7 +143,11 @@ in
 
   config = {
 
-    environment.systemPackages = requiredPackages ++ config.environment.defaultPackages;
+    environment.systemPackages = requiredPackages ++ (lib.remove pkgs.nano config.environment.defaultPackages);
+
+    programs.nano = {
+      enable = mkDefault (builtins.elem pkgs.nano config.environment.defaultPackages);
+    };
 
     environment.pathsToLink =
       [ "/bin"
```
The priority won't be lower than a user added nano, but maybe it'll be ok

Reply:
> nano is a defaultPackage, so this could break some people's setups

The file is only recently created since the refactor of the module. Before it wasn't there, otherwise I wouldn't have noticed it on the update.